chapter04/rpi/ex2.pl

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+sub check_items_for_all {
+  my ( $all ) = @_;
+
+  foreach $name ( keys %$all ) {
+    check_required_items( $name, $all->{name} );
```

Comment:
This passes an arrayref into check_required_items(), which I'd usually recommend, but the function in the chapter is expecting an array of items.